### DISCUSSION TITLE
Use of hires image instead of lowres in SpatialDimPlot

### DISCUSSION THREAD

**Discussion by Lisa-Salhi** [satijalab/seurat#4833] [Q&A]
Hello,

How can I overlap ​the spots on the highres image instead of the lowres one?
When I do `SpatialDimPlot(allsamples.merged, label.size = 3, alpha = 0.1, group.by = "seurat_clusters", images = "slice1")`, the image shown is the lowres one.

The Seurat version I use is 4.0.1

Best,
Lisa

**Comment by ggruenhagen3** [comment]
Hi @Lisa-Salhi,

If you have not created a Seurat object or are willing to start from the beginning, you can follow these steps:
`img = Read10X_Image("/path/to/spaceranger_output/outs/spatial/", image.name = "tissue_hires_image.png")`
`new.object = Load10X_Spatial("/path/to/spaceranger_output/outs/", image = img)`
`new.object@images@slice1@scale.factors$lowres = new.object@images@slice1@scale.factors$hires`
__________________________________________________________________________________________________
If you already have a Seurat object and don't feel like creating a new one just to use the high-res image, follow these steps:
`lowres = object@images$slice1`
`hires = Read10X_Image("/path/to/spaceranger_output/outs/spatial/", image.name = "tissue_hires_image.png")`

If you removed any cells or renamed any of the spots in anyway, the same modifications need to be repeated in the `hires@coordinates`. For example, I added the prefix "b2d_" to all of my spots and I removed a few during quality control. I can easily make those changes to `hires@coordinates` so that way Seurat doesn't get confused. If you don't do this step, you will get an error when you plot that looks something like this: _Error in data[[features[j]]] : subscript out of bounds_.
`rownames(hires@coordinates) = paste0("b2d_", rownames(hires@coordinates))`
`hires@coordinates = hires@coordinates[rownames(lowres@coordinates),]`

We need to use the hires scalefactor instead of the lowres scalefactor. You can see my [solution](https://github.com/satijalab/seurat/issues/5614#issuecomment-1194555472) for a problem in another thread for more details.
`hires@scale.factors$lowres = hires@scale.factors$hires`

Set the assay and key to the same as the lowres (I'm not sure if this is necessary or not).
`hires@assay = lowres@assay`
`hires@key = lowres@key`

Finally, add the image into the object and everything should be good! 
`new.object@images$slice1 = hires`

Hope this helps!

**Reply by sciencepeak** [reply]
Under the case that "you have not created a Seurat object or are willing to start from the beginning, you can follow these steps:"
```r
img = Read10X_Image("/path/to/spaceranger_output/outs/spatial/", image.name = "tissue_hires_image.png")
img@scale.factors$lowres = img@scale.factors$hires # it is better to set the scale factors this way.
new.object = Load10X_Spatial("/path/to/spaceranger_output/outs/", image = img)
```
Thereby, you can avoid this statement:
```r
new.object@images@slice1@scale.factors$lowres = new.object@images@slice1@scale.factors$hires
```
because this statement use the slice1 that is a specified sample name, it is hard to process multiple samples in a programming style. After all, the sample names are stored in a variable.

Thanks for the preceding great question and answer, which are very helpful.

**Reply by ggruenhagen3** [reply]
You can always change the name of the images you store in your object with names(object@images). I programmatically store high resolution images for 16 samples.
